### PR TITLE
fix: VOL-5103 limit the access of read only users to certain data and  actions

### PR DESCRIPTION
**Pull request by ilindsay**
## Description

Limit the access of read only users to certain data and actions

Related issue: [VOL-5103](https://dvsa.atlassian.net/browse/VOL-5103)

### CODE REVIEW

File: Common/src/Common/Controller/Lva/Application/AbstractTypeOfLicenceController.php [hunk 1]
```diff
@@ -75,6 +75,8 @@ public function indexAction()
             return $this->notFoundAction();
         }
 
+        $isInternalReadOnly = $this->isInternalReadOnly();
```
Comment (JoshuaLicense) [inline Common/src/Common/Controller/Lva/Application/AbstractTypeOfLicenceController.php:78]: Is this used?

File: Common/src/Common/Controller/Lva/AbstractController.php [hunk 2]
```diff
@@ -381,17 +382,16 @@ protected function getLicenceId($applicationId = null)
...
     {
         return $this->location === self::LOC_EXTERNAL;
     }
 
     /**
-     * Return true if the current internal user has read only permissions
-     *
-     * @return bool
+     * Have left this in place for now as the number of controllers extending it made removal impractical
+     * @see Permission for the preferred way to access this logic via dependency injection
```
Comment (JoshuaLicense) [inline Common/src/Common/Controller/Lva/AbstractController.php:392]: Should we add `@deprecated` annotation and point to the new `Permission` service here?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]
Few minor things, but looks good!

File: Common/src/Common/Controller/Lva/AbstractController.php (hunk 2 above)
Comment (ilindsay) [reply Common/src/Common/Controller/Lva/AbstractController.php]: Yes - added the annotation. The comment did point to the permission service (in the sense the IDE could follow it) but have swapped to a FQCN so it's more readable to human eye.

**Review by JoshuaLicense — APPROVED** [review]

**Review by fibble — APPROVED** [review]

**Pull request merged**